### PR TITLE
Rework the parse of the Authorization/WWW-Authenticate header

### PR DESCRIPTION
**Pull request by nbeyer**
handle both quoted strings and tokens per https://tools.ietf.org/html/rfc7235#appendix-C.

### CODE REVIEW

File: lib/cerner/oauth1a/protocol.rb [hunk 1]
```diff
@@ -57,10 +57,18 @@ def self.parse_authorization_header(value)
         value = value.strip
         return params unless value.size > 6 && value[0..5].casecmp?('OAuth ')
 
-        value.scan(/([^,\s=]*)=\"([^\"]*)\"/).each do |pair|
-          k = URI.decode_www_form_component(pair[0])
-          v = URI.decode_www_form_component(pair[1])
-          params[k.to_sym] = v
+        # trim off 'OAuth ' prefix
```
Comment (venki09) [inline lib/cerner/oauth1a/protocol.rb:60]: Should we also add an example of token to the method doc? https://github.com/cerner/cerner-oauth1a/blob/716d43405cd16a8a0dd56ffa91a1dbfc70a52cdf/lib/cerner/oauth1a/protocol.rb#L48-L50

Comment (nbeyer) [reply lib/cerner/oauth1a/protocol.rb:64]: Added at b9aecce.

**Comment by AnthonyRoss** [comment]
@nbeyer is this library compatible with ruby-2.4.x anymore? If not that is probably fine it just looks like the build is still checking for that

**Comment by nbeyer** [comment]
@AnthonyRoss Yeah, the 2.5.x version is supposed to be compatible with Ruby 2.4. I just fixed the issue, which was also failing Ruby 2.5. I was using the infinite range syntax (i.e. `var[1..]`), which was added in Ruby 2.6. It should be fixed now.

**Pull request merged**